internal/server/server.go

```diff
@@ -0,0 +1,108 @@
+package server
+
+import (
+	"context"
+	"net/http"
+	"net/http/pprof"
+	"time"
+
+	"github.com/observatorium/observatorium/internal/proxy"
+	"github.com/observatorium/observatorium/prober"
+
+	"github.com/go-chi/chi"
+	"github.com/go-chi/chi/middleware"
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+type Server struct {
+	logger log.Logger
+	prober *prober.Prober
+	srv    *http.Server
+
+	opts options
+}
+
+func New(logger log.Logger, reg *prometheus.Registry, opts ...Option) Server {
+	options := options{
+		gracePeriod: 5 * time.Second,
+		profile:     false,
+	}
+
+	for _, o := range opts {
+		o.apply(&options)
+	}
+
+	ins := newInstrumentationMiddleware(reg)
+	p := prober.New(logger)
+
+	r := chi.NewRouter()
+	r.Use(middleware.RequestID)
+	r.Use(middleware.RealIP)
+	r.Use(middleware.Recoverer)
+
+	registerProber(r, p)
+	r.Get("/metrics", func(w http.ResponseWriter, r *http.Request) {
+		promhttp.InstrumentMetricHandler(reg, promhttp.HandlerFor(reg, promhttp.HandlerOpts{})).ServeHTTP(w, r)
+	})
+
+	r.Get("/prometheus/query",
```

Comment:
I think we'd want this to be versioned and technology independent, so this should be "/v1/metrics/query" no? Plus this needs query and query_range.

Reply:
👍I have overlooked those paths, I'll make the necessary changes.